test/integ/pandas/test_arrow_pandas.py

```diff
@@ -583,6 +583,54 @@ def test_timestampltz(conn_cnx, scale, timezone):
         finish(conn, table)
 
 
+@pytest.mark.skipif(
+    not installed_pandas or no_arrow_iterator_ext,
+    reason="arrow_iterator extension is not built, or pandas is missing.",
+)
+def test_vector(conn_cnx, is_public_test):
+    if is_public_test:
+        pytest.xfail(
+            reason="This feature hasn't been rolled out for public Snowflake deployments yet."
+        )
+    tests = [
+        (
+            "vector(int,3)",
+            [
+                "NULL",
+                "[1,2,3]::vector(int,3)",
```

Comment:
Just out of my curiosity, do we allow `null` as a vector element?

Reply:
Individual vector elements can't be null, but the vector as a whole could be